### ISSUE TITLE
Recognize widget types as strings

### ISSUE THREAD

**Issue by sofroniewn** [napari/magicgui#47]
Similar to #46, I'd like to be able to pass widget types as strings to avoid additional dependencies, for example from the magic gui parameter sweep example https://magicgui.readthedocs.io/en/latest/examples/napari_parameter_sweep/

```python
    @magicgui(
        auto_call=True,
        sigma={"widget_type": 'magicgui._qt.widgets.QDoubleSlider', "maximum": 6, "fixedWidth": 400},
        mode={"choices": ["reflect", "constant", "nearest", "mirror", "wrap"]},
    )
```

but right now I get the following error
```python-traceback
  File "/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/core.py", line 652, in __init__
    **param_options,
  File "/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/core.py", line 190, in __init__
    dtype=(None if param.annotation is param.empty else param.annotation),
  File "/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/core.py", line 360, in set_widget
    widget = api.make_widget(WidgetType, name=name, parent=self, **_options)
  File "/Users/nsofroniew/opt/anaconda3/lib/python3.7/site-packages/magicgui/_qt/_qt.py", line 188, in make_widget
    widget = WidgetType(parent=parent)
TypeError: 'str' object is not callable
```

Additionally I'm not sure if this will be changed in #43, but it would be nice not to have to use a private import or a `Q` specific syntax to get a DoubleSlider.

**Comment by tlambert03** [comment]
Yeah, this is one of the main things changed in #43.  magicgui acquires its own backend-independent widgets, so this specific example will become:
```python
from magicgui.widgets import FloatSlider
@magicgui(sigma={"widget_type": FloatSlider, "maximum": 6})
...

# and it will also work as a string
sigma={"widget_type": "slider", "maximum": 6}
```

losing support, for now, for Qt specific things like `fixedWidth`...

but gaining the ability to just directly create a backend-independent widget:

`widget = FloatSlider()`

**Comment by sofroniewn** [comment]
Love it, I definitely think moving in that direction makes sense

**Comment by sofroniewn** [comment]
I think this can now be closed with the merge of #43, very exciting!
